contracts/RedemptionVaultWithUSTB.sol

```diff
@@ -0,0 +1,177 @@
+     * Burns mToken from the user.
+     * Transfers fee in mToken to feeReceiver
+     * Transfers tokenOut to user.
+     * @param tokenOut token out address
+     * @param amountMTokenIn amount of mToken to redeem
+     * @param minReceiveAmount minimum expected amount of tokenOut to receive (decimals 18)
+     */
+    function redeemInstant(
+        address tokenOut,
+        uint256 amountMTokenIn,
+        uint256 minReceiveAmount
+    )
+        external
+        override
+        whenFnNotPaused(this.redeemInstant.selector)
+        onlyGreenlisted(msg.sender)
+        onlyNotBlacklisted(msg.sender)
+        onlyNotSanctioned(msg.sender)
+    {
+        address user = msg.sender;
+
+        (
+            uint256 feeAmount,
+            uint256 amountMTokenWithoutFee
+        ) = _calcAndValidateRedeem(user, tokenOut, amountMTokenIn, true, false);
+
+        _requireAndUpdateLimit(amountMTokenIn);
+
+        uint256 tokenDecimals = _tokenDecimals(tokenOut);
+
+        uint256 amountMTokenInCopy = amountMTokenIn;
+        address tokenOutCopy = tokenOut;
+        uint256 minReceiveAmountCopy = minReceiveAmount;
+
+        (uint256 amountMTokenInUsd, uint256 mTokenRate) = _convertMTokenToUsd(
+            amountMTokenInCopy
+        );
+        (uint256 amountTokenOut, uint256 tokenOutRate) = _convertUsdToToken(
+            amountMTokenInUsd,
+            tokenOutCopy
+        );
+
+        _requireAndUpdateAllowance(tokenOutCopy, amountTokenOut);
+
+        mToken.burn(user, amountMTokenWithoutFee);
+        if (feeAmount > 0)
+            _tokenTransferFromUser(address(mToken), feeReceiver, feeAmount, 18);
+
+        uint256 amountTokenOutWithoutFee = (amountMTokenWithoutFee *
+            mTokenRate) / tokenOutRate;
+
+        require(
+            amountTokenOutWithoutFee >= minReceiveAmountCopy,
+            "RVU: minReceiveAmount > actual"
+        );
+
+        uint256 amountTokenOutWithoutFeeFrom18 = amountTokenOutWithoutFee
+            .convertFromBase18(tokenDecimals);
+
+        _checkAndRedeemUSTB(tokenOutCopy, amountTokenOutWithoutFeeFrom18);
+
+        _tokenTransferToUser(
+            tokenOutCopy,
+            user,
+            amountTokenOutWithoutFeeFrom18.convertToBase18(tokenDecimals),
+            tokenDecimals
+        );
+
+        emit RedeemInstant(
+            user,
+            tokenOutCopy,
+            amountMTokenInCopy,
+            feeAmount,
+            amountTokenOutWithoutFee
+        );
+    }
+
+    /**
+     * @notice Check if contract has enough USDC balance for redeem
+     * if not, trigger USTB redemption flow to redeem exactly the missing amount
+     * @param tokenOut tokenOut address
+     * @param amountTokenOut amount of tokenOut needed
+     */
+    function _checkAndRedeemUSTB(
+        address tokenOut,
+        uint256 amountTokenOut
+    ) internal {
+        require(tokenOut == ustbRedemption.USDC(), "RVU: invalid token");
```

Comment:
in this case it will always revert if token out is not usdc, even if ustb redeem wont be triggered